### ISSUE TITLE
Flutter web does not render arabic (RTL) characters properly when HTML renderer used.

### ISSUE THREAD

**Issue by rebindev** [flutter/flutter#133374]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1-Create flutter project.
2-Use text widget with arabic text.
3-flutter run -d chrome --**web-renderer html**
I tested on flutter 3.10.6 and 13.3.1 with same results and both on mobile and desktop browsers.

### Expected results

Expected those Arabic characters to be rendered properly like:

سڵاو باشی

### Actual results

Got the chracters in reverse.

<img width="1200" alt="Screenshot 2023-08-26 at 3 49 35 AM" src="https://github.com/flutter/flutter/assets/66472329/8b1cc6b9-b784-45d7-a1b8-444e4ed1c62c">

### Code sample

<details><summary>Code sample</summary>

```dart
[Widget build(BuildContext context) {
    print('built dashboard');

    return Scaffold(
      body: Column(
        children: [Text('سڵاو باشی')],
      ),
    );
  }]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.4 22F66 darwin-arm64, locale en-IQ)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.81.1)
[✓] Connected device (2 available)
    ! Device 1WMHHA615N2015 is not authorized.
      You might need to check your device for an authorization dialog.
[✓] Network resources
]
```

</details>

**Comment by darshankawar** [comment]
@rebindev 
Can you check if your case resembles https://github.com/flutter/flutter/issues/122153 ?

**Comment by rebindev** [comment]
@darshankawar not really. Mine the characters of every word are reversed for example lets assume this is arabic text. 

We go 

when HTML render used the text rendered as .. ew og. 

Currently my work around is using:
TextStyle(
letterSpacing: 0,
)
 Using this style there is no issues.

**Comment by rebindev** [comment]
@darshankawar Yes my issue is different. The one you mentioned have some problem ordering words.. but mine is absolutely unreadable and the characters are separated and reversed.

**Comment by rebindev** [comment]
@darshankawar [This issue](https://github.com/flutter/flutter/issues/108608) is similar to mine.

**Comment by darshankawar** [comment]
Thanks for the update @rebindev 
I tried the code sample you shared in original post and also from the code sample from above issue, ran it in web using html renderer with latest stable and master version and observed that I saw correct result:


<img width="401" alt="Screenshot 2023-08-29 at 10 48 57 AM" src="https://github.com/flutter/flutter/assets/67046386/d5e2f456-46c5-4265-9aab-f1a1b2b4ca37">
<img width="380" alt="Screenshot 2023-08-29 at 10 50 21 AM" src="https://github.com/flutter/flutter/assets/67046386/f237d69b-9de8-4adf-9cb8-93c7fec4327f">
<img width="450" alt="Screenshot 2023-08-29 at 10 59 52 AM" src="https://github.com/flutter/flutter/assets/67046386/6e9ae77a-f905-4c60-b748-21e7c052bd73">

**Comment by PuzzleTakX** [comment]
Use this rendering : 
flutter build web --no-sound-null-safety --web-renderer canvaskit

**Comment by PuzzleTakX** [comment]
I had this problem before, if the output is taken with this code, the problem will be solved code
`flutter build web --no-sound-null-safety --web-renderer canvaskit`
